javagen/src/main/java/com/azure/autorest/mapper/ObjectMapper.java

```diff
@@ -42,23 +42,33 @@ public IType map(ObjectSchema compositeType) {
         }
 
         if (result == null) {
-            String classPackage;
-            if (settings.isCustomType(compositeType.getLanguage().getJava().getName())) {
-                classPackage = settings.getPackage(settings.getCustomTypesSubpackage());
-            } else if (!settings.isFluent()) {
-                classPackage = settings.getPackage(settings.getModelsSubpackage());
+            if (isPlainObject(compositeType)) {
+                result = ClassType.Object;
+            } else {
+                String classPackage;
+                if (settings.isCustomType(compositeType.getLanguage().getJava().getName())) {
+                    classPackage = settings.getPackage(settings.getCustomTypesSubpackage());
+                } else if (!settings.isFluent()) {
+                    classPackage = settings.getPackage(settings.getModelsSubpackage());
+                }
+//                else if (compositeType.IsInnerModel)
+//                {
+//                    classPackage = settings.getPackage(settings.getImplementationSubpackage());
+//                }
+                else {
+                    classPackage = settings.getPackage();
+                }
+                result = new ClassType(classPackage, compositeType.getLanguage().getJava().getName(), null, compositeType.getExtensions(), false/*compositeType.IsInnerModel*/);
             }
-//            else if (compositeType.IsInnerModel)
-//            {
-//                classPackage = settings.getPackage(settings.getImplementationSubpackage());
-//            }
-            else {
-                classPackage = settings.getPackage();
-            }
-            result = new ClassType(classPackage, compositeType.getLanguage().getJava().getName(), null, compositeType.getExtensions(), false/*compositeType.IsInnerModel*/);
         }
 
         parsed.put(compositeType, result);
         return result;
     }
+
+    public static boolean isPlainObject(ObjectSchema compositeType) {
+        return compositeType.getProperties().isEmpty() && compositeType.getDiscriminator() == null
+                && compositeType.getParents() == null && compositeType.getChildren() == null
+                && (compositeType.getExtensions() == null || compositeType.getExtensions().getXmsEnum() == null);
```

Comment:
I am not sure whether this is complete or not.

No parent, no child, no property, no polymorphy, no enum (not sure this applies not not. seen type: object with x-ms-enum in swagger).

Reply:
Swagger https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Resources/stable/2019-08-01/resources.json#L5352-L5355

We may able to remove this in 2024